### PR TITLE
[Telemetry] Adding x-ms-user-agent in Dataverse calls to check the client  surface (Vscode Web/Desktop)

### PR DESCRIPTION
**Pull request by BidishaMS**
[Telemetry] Adding x-ms-user-agent in Dataverse calls to check the client  surface (Vscode Web/Desktop)

### CODE REVIEW

**Review by amitjoshi438 — APPROVED** [review]

**Review by amitjoshi438 — APPROVED** [review]

File: src/common/copilot/dataverseMetadata.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ export async function getEntityColumns(entityName: string, orgUrl: string, apiTo
             headers: {
                 'Content-Type': "application/json",
                 Authorization: `Bearer ${apiToken}`,
+                "x-ms-user-agent": "Vscode-Desktop"
```
Comment (joem-msft) [inline src/common/copilot/dataverseMetadata.ts:28]: This is a great idea.
The expected format of this string is the same as the [User-Agent header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent). Namely:
`x-ms-user-agent: <product> / <product-version> <comment>`

1. Let's add the version of the product too. It must be the version of the 'product'
1. For 'product', Should it actually be the name or id of the extension? If not, and we want it to be the VSCode product, can we set the product value to something we get from the VSCode SDK instead?
2. In the 'comment' part, is where extra information is usually set. e.g. the desktop vs web would be more appropriate there. e.g. `VSCode/1.88.1 (Desktop; other; another)`

File: src/common/copilot/dataverseMetadata.ts [hunk 2]
```diff
@@ -57,6 +58,7 @@ export async function getFormXml(entityName: string, formName: string,  orgUrl:
...
```
Comment (joem-msft) [inline src/common/copilot/dataverseMetadata.ts:61]: Looks like this header constant is being used in multiple places. Consider moving it's value to a reusable function, with different implementation per Desktop/Web.
e.g. `composeXMsUserAgent`

**Review by joem-msft — CHANGES_REQUESTED** [review]
only needing the product version to be added to the x-ms-user-agent header values.
all other comments are recommendations.

File: src/common/copilot/dataverseMetadata.ts (hunk 2 above)
Comment (tyaginidhi) [reply src/common/copilot/dataverseMetadata.ts:61]: +1 to this comment  @BidishaMS

**Review by tyaginidhi — CHANGES_REQUESTED** [review]

File: src/common/copilot/dataverseMetadata.ts (hunk 1 above)
Comment (BidishaMS) [reply src/common/copilot/dataverseMetadata.ts]: Done

File: src/common/copilot/dataverseMetadata.ts (hunk 2 above)
Comment (BidishaMS) [reply src/common/copilot/dataverseMetadata.ts]: Done

File: src/common/Utils.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 
 import * as vscode from "vscode";
-import { EXTENSION_ID, SETTINGS_EXPERIMENTAL_STORE_NAME } from "../client/constants";
+import { EXTENSION_ID,EXTENSION_NAME, SETTINGS_EXPERIMENTAL_STORE_NAME } from "../client/constants";
```
Comment (tyaginidhi) [inline src/common/Utils.ts:8]: nit: spacing not accurate

**Review by tyaginidhi — APPROVED** [review]

File: src/common/Utils.ts (hunk 3 above)
Comment (BidishaMS) [reply src/common/Utils.ts]: fixed

File: src/common/Utils.ts [hunk 4]
```diff
@@ -123,3 +123,12 @@ export function isCustomTelemetryEnabled():boolean {
         .get(CUSTOM_TELEMETRY_FOR_POWER_PAGES_SETTING_NAME);
     return isCustomTelemetryEnabled as boolean;
 }
+
+export function getUserAgent(): string {
+    const userAgent = "{product} / {product-version} {comment}";
```
Comment (joem-msft) [inline src/common/Utils.ts:128]: there should not be any space between the product, '/' char and version. The space after the version is significant.
"{product}/{product-version} {comment}"

File: src/common/Utils.ts [hunk 5]
```diff
@@ -123,3 +123,12 @@ export function isCustomTelemetryEnabled():boolean {
...
+
+    return userAgent
+        .replace("{product}", EXTENSION_NAME)
+        .replace("{product-version}", getExtensionVersion())
+        .replace("{comment}", getExtensionType()+'; '+getExtensionVersion());
```
Comment (joem-msft) [inline src/common/Utils.ts:133]: See how browsers do this. The comment, if specified is usually in parentheses. This makes it so spaces can be inside the comment.
i.e. `"myproduct/0.1.0-dev (extVer:5.0; otherThing:value; flagValue)"`

Comment (joem-msft) [inline src/common/Utils.ts:133]: Looks like the extension version is there twice. Remove from the comment?

**Review by joem-msft — CHANGES_REQUESTED** [review]

File: src/common/Utils.ts (hunk 5 above)
Comment (BidishaMS) [reply src/common/Utils.ts]: removed

Comment (BidishaMS) [reply src/common/Utils.ts]: put it in parenthesis

**Review by joem-msft — APPROVED** [review]

**Pull request merged**